### PR TITLE
Hardening for SonarLint

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

**Comment by henryju** [comment]
@dbmeneses I would like your opinion on the first commit since it is an attempt to fix this very old issue: https://jira.sonarsource.com/browse/SSLRSQBR-72 and since it is about Thread, I would like your opinion.

**Comment by dbmeneses** [comment]
LGTM!

File: commons/src/main/java/org/sonarsource/analyzer/commons/ProgressReport.java [hunk 1]
```diff
@@ -54,22 +55,33 @@ public ProgressReport(String threadName, long period) {
...
-      if (success) {
-        log(count + "/" + count + " source files have been " + adjective);
-      }
+    if (success) {
+      log(count + "/" + count + " source " + pluralizeFile(count) + " " + pluralizeHas(count) + " been " + adjective);
+    }
+  }
+
+  private static String pluralizeFile(long count) {
+    if (count == 1) {
```
Comment (jblievremont) [inline commons/src/main/java/org/sonarsource/analyzer/commons/ProgressReport.java:74]: Nitpick (and same in `pluralizeHas`):
```suggestion
    if (count == 1L) {
```

**Review by jblievremont — APPROVED** [review]
:+1:

File: commons/src/main/java/org/sonarsource/analyzer/commons/ProgressReport.java [hunk 2]
```diff
@@ -92,14 +103,20 @@ public synchronized void nextFile() {
...
+    join();
   }
 
-  public void join() throws InterruptedException {
-    thread.join();
+  private void join() {
+    try {
+      thread.join();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
```
Comment (m-g-sonar) [inline commons/src/main/java/org/sonarsource/analyzer/commons/ProgressReport.java:118]: These two lines are apparently not covered. Would it be possible to try to check them?
I feel it's important as it propagates the interruption.

**Comment by m-g-sonar** [comment]
@henryju If you can fix the coverage, I would appreciate it. I also think that small fix suggested by @jblievremont are good improvements.
Otherwise, LGTM!

File: commons/src/main/java/org/sonarsource/analyzer/commons/ProgressReport.java (hunk 2 above)
Comment (henryju) [reply commons/src/main/java/org/sonarsource/analyzer/commons/ProgressReport.java:118]: Not very simple to test, but here it is.

**Review by m-g-sonar — APPROVED** [review]
LGTM! thanks for the changes!

**Pull request merged**